spamFilter.go

```diff
@@ -124,9 +143,23 @@ func isSpam(newMail *os.File, probMap map[string]float64) bool {
 		wordProbSlice[i].interest = math.Abs(0.5 - prob)
 		i++
 	}
-	quickSort(wordProbSlice, 0, len(wordProbSlice) - 1)
-	
-	
+	//quickSort(wordProbSlice, 0, len(wordProbSlice) - 1)
+	sort.Sort(ByInterest(wordProbSlice))
+	mostInteresting := wordProbSlice[len(wordProbSlice) - 15:]
+	var probProd, invProbProd float64 = 1.0
+	for _, v := range mostInteresting {
+		probProd = v.probability * probProd
+		invProb := 1 - v.probability
+		invProbProd = invProb * invProbProd
+	}	
+	combProd := probProd/(probProd + invProbProd)
+	var mailIsSpam bool
+	if combProd >= 0.9 {
+		mailIsSpam = true
+	} else {
+		mailIsSpam = false
+	}
+	return mailIsSpam
```

Comment:
```suggestion
return combProd >= 0.9
```

See how this is equivalent?

Reply:
You could even inline the expression above; `return probProd/(probProd + invProbProd) >= 0.9`.